### PR TITLE
User settings: new backend system

### PR DESCRIPTION
**Pull request by vinzscam**
## Hey, I just made a Pull Request!

Migrated user settings to the new backend system

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/frontend-app-api/src/wiring/createApp.tsx [hunk 1]
```diff
@@ -57,9 +57,14 @@ import {
...
   ApiProvider,
   ApiResolver,
-  AppThemeSelector,
 } from '@backstage/core-app-api';
 
+import {
+  AppLanguageSelector,
+  AppThemeSelector,
+  I18nextTranslationApi,
+} from '@backstage/core-app-api/alpha';
```
Comment (Rugvip) [inline packages/frontend-app-api/src/wiring/createApp.tsx:66]: Let's use internal relative imports for now like the things below, don't really want to export these at this point.

File: plugins/user-settings/src/alpha.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * @alpha
+ */
+export const userSettingsRouteRef = createRouteRef({
+  id: 'plugin.user-settings.page',
+});
+
+/**
+ * @alpha
+ */
+export const UserSettingsPage = createPageExtension({
```
Comment (Rugvip) [inline plugins/user-settings/src/alpha.tsx:38]: Let's skip exporting the actual extensions, just export the plugin

File: plugins/user-settings/src/alpha.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  id: 'plugin.user-settings.page',
+  defaultPath: '/settings/*',
```
Comment (Rugvip) [inline plugins/user-settings/src/alpha.tsx:40]: Skip the `/*` suffix here, it shouldn't need to be here, we need to fix the core routes instead.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice 😁 

Just a couple of nits

File: .changeset/heavy-plants-doubt.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-user-settings': patch
+---
+
+Added experimental support for declarative integration
```
Comment (Rugvip) [inline .changeset/heavy-plants-doubt.md:5]:
```suggestion
Added experimental support for declarative integration via the `/alpha` export.
```

I like calling this out just to make it clear where the change is being made so users wouldn't go looking for it in the wrong place

File: packages/app-next/src/App.tsx [hunk 5]
```diff
@@ -48,8 +55,26 @@ TODO:
...
 /* app.tsx */
 
+const fakePlugin = createPlugin({
+  id: 'fake',
+  extensions: [
+    createPageExtension({
+      id: 'catalog:entity',
+      defaultPath: '/fakeentity',
+      routeRef: entityRouteRef,
+      loader: async () => <div>hello 😅</div>,
```
Comment (Rugvip) [inline packages/app-next/src/App.tsx:65]: Is the intention to keep this around? 😅

File: plugins/user-settings/src/alpha.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import { createRouteRef } from '@backstage/core-plugin-api';
+import {
+  coreExtensionData,
+  createExtensionInput,
+  createPageExtension,
+  createPlugin,
+} from '@backstage/frontend-plugin-api';
```
Comment (Rugvip) [inline plugins/user-settings/src/alpha.tsx:7]: woah

**Review by Rugvip — APPROVED** [review]
Nice 😁 

Just a bit of cleanup to be done, but approved ahead of that

File: packages/app-next/src/App.tsx (hunk 5 above)
Comment (vinzscam) [reply packages/app-next/src/App.tsx:65]: it's actually necessary as user settings plugin depends on `entityRouteRef`. I can polish a bit though 😅

File: .changeset/heavy-plants-doubt.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-user-settings': patch
+---
+
+Added experimental support for declarative integration via the `/alpha` export.
```
Comment (Rugvip) [inline .changeset/heavy-plants-doubt.md:5]:
```suggestion
Added experimental support for declarative integration via the `/alpha` subpath.
```

Had this wrong in the first suggestion :p

**Pull request merged**